cloud-controller-manager/do/loadbalancers.go

```diff
@@ -144,10 +157,36 @@ func (l *loadbalancers) GetLoadBalancer(ctx context.Context, clusterName string,
 // EnsureLoadBalancer will not modify service or nodes.
 func (l *loadbalancers) EnsureLoadBalancer(ctx context.Context, clusterName string, service *v1.Service, nodes []*v1.Node) (*v1.LoadBalancerStatus, error) {
 	lbStatus, exists, err := l.GetLoadBalancer(ctx, clusterName, service)
-	if err != nil {
+	if _, ok := err.(*errLBTimeout); !ok && err != nil {
 		return nil, err
 	}
 
+	if _, ok := err.(*errLBTimeout); ok && exists {
+		// we should attempt to delete load balancers where state is not active after the provided timeout
+		// and if it has not been given an IP yet. Load balancers in an error state that already have an IP address
+		// should be manually recreated.
+
+		// we should only delete this load balancer, if a previous IP was not assigned to it.
+		// this ensures load balancerse with an already assigned IP aren't accidentally deleted.
+		if len(service.Status.LoadBalancer.Ingress) > 0 {
+			return nil, err
```

Comment:
How about annotating the error to indicate to the user that manual intervention is required? Something like

```
return nil, fmt.Errorf("cannot reconcile load-balancer in error state having IP address assigned already, needs to be recreated manually: %s", err)
```

Reply:
good idea